meta-mender-core/recipes-mender/mender/files/0001-Allow-mulitple-digit-partition-numbers.patch

```diff
@@ -0,0 +1,91 @@
+From 62378435984a763671a2d0347b857766b9b0c611 Mon Sep 17 00:00:00 2001
+From: Drew Moseley <drew.moseley@northern.tech>
+Date: Mon, 22 Jan 2018 15:51:40 -0500
+Subject: [PATCH] Allow mulitple digit partition numbers.
+
+Signed-off-by: Drew Moseley <drew.moseley@northern.tech>
+---
+ src/github.com/mendersoftware/mender/device.go | 30 ++++++++++++++++++++----------
+ 1 file changed, 20 insertions(+), 10 deletions(-)
+
+diff --git a/src/github.com/mendersoftware/mender/device.go b/src/github.com/mendersoftware/mender/device.go
+index 9b6ccca..8cca1b1 100644
+--- a/src/github.com/mendersoftware/mender/device.go
++++ b/src/github.com/mendersoftware/mender/device.go
+@@ -14,6 +14,8 @@
+ package main
+ 
+ import (
++	"strings"
++	"fmt"
+ 	"io"
+ 	"path/filepath"
+ 	"strconv"
+@@ -57,13 +59,13 @@ func (d *device) Reboot() error {
+ 
+ func (d *device) SwapPartitions() error {
+ 	// first get inactive partition
+-	inactivePartition, err := d.getInactivePartition()
++	inactivePartition, inactivePartitionHex, err := d.getInactivePartition()
+ 	if err != nil {
+ 		return err
+ 	}
+ 	log.Infof("setting partition for rollback: %s", inactivePartition)
+ 
+-	err = d.WriteEnv(BootVars{"mender_boot_part": inactivePartition, "upgrade_available": "0"})
++	err = d.WriteEnv(BootVars{"mender_boot_part": inactivePartition, "mender_boot_part_hex": inactivePartitionHex, "upgrade_available": "0"})
+ 	if err != nil {
+ 		return err
+ 	}
+@@ -137,32 +139,40 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
+ 	return err
+ }
+ 
+-func (d *device) getInactivePartition() (string, error) {
++func (d *device) getInactivePartition() (string, string, error) {
+ 	inactivePartition, err := d.GetInactive()
+ 	if err != nil {
+-		return "", errors.New("Error obtaining inactive partition: " + err.Error())
++		return "", "", errors.New("Error obtaining inactive partition: " + err.Error())
+ 	}
+ 
+ 	log.Debugf("Marking inactive partition (%s) as the new boot candidate.", inactivePartition)
+ 
+-	partitionNumber := inactivePartition[len(inactivePartition)-1:]
+-	if _, err := strconv.Atoi(partitionNumber); err != nil {
+-		return "", errors.New("Invalid inactive partition: " + inactivePartition)
++	partitionSlice := strings.SplitAfter(inactivePartition, "p")
```

Comment:
This won't work for `/dev/sda2` type of partitions, which is likely to be common on larger boards, such as Intel. I think you will have to count digits backwards from the end of the string.

Suggestion (untested):

```
partitionNumberDecStr := inactivePartition[len(strings.TrimRight(inactivePartition, "0123456789")):]
```

which gives you the final digits.